src/org/openstreetmap/josm/plugins/fieldpapers/FieldPapersAddLayerAction.java

```diff
@@ -28,10 +35,29 @@ public FieldPapersAddLayerAction() {
     }
 
     public void actionPerformed(ActionEvent e) {
-        String url = JOptionPane.showInputDialog(Main.parent,
-            tr("Enter a fieldpapers.org snapshot URL"),
-                Main.pref.get("fieldpapers.last-used-id"));
+        JPanel panel = new JPanel(new GridBagLayout());
+
+        // snapshot URL selection
+        panel.add(new JLabel(tr("Enter a fieldpapers.org snapshot URL")), GBC.eol());
+        JosmTextField snapshotAddress = new JosmTextField(Main.pref.get("fieldpapers.last-used-id"));
+        snapshotAddress.setToolTipText(tr("Enter an URL from where scanned map should be downloaded"));
+        panel.add(snapshotAddress, GBC.eop().fill(GBC.BOTH));
+
+        ExtendedDialog dialog = new ExtendedDialog(Main.parent,
+                tr("Download Snapshot"),
+                tr("Download URL"), tr("Cancel"))
```

Comment:
The only thing I would change would be "Download **from** URL" instead of "Download URL", or you could change the description to "Enter a fieldpapers.org snapshot URL to download from" and name the button simply "Download".
But otherwise I like your changes, especially the reuse of components that are already there in JOSM core and that you splitted the method into two for displaying the dialog and downloading from the URL.